### PR TITLE
util/types: fix order by binary

### PR DESCRIPTION
**Pull request by iamxy**
In Orderby Clause, as using BINARY operator to cast the following operand to a binary string, here meet an error. Improved IsOrderedType() to handle it.

### CODE REVIEW

File: util/types/etc.go [hunk 1]
```diff
@@ -174,13 +174,23 @@ func collateDesc(a, b []interface{}) int {
 // IsOrderedType returns a boolean
 // whether the type of y can be used by order by.
 func IsOrderedType(v interface{}) (r bool) {
-	switch v.(type) {
+	switch x := v.(type) {
```
Comment (shenli) [inline util/types/etc.go]: You can use util/types.RawData() to get real data.

Comment (iamxy) [inline util/types/etc.go]: Got it！

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**